tests/lib/tools/feature_extractor.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+
+import argparse
+from collections import defaultdict
+import json
+
+
+# This will be removed
+class AllFeature:
+    name = "all"
+    parent = "all"
+    
+    @staticmethod
+    def maybe_add_feature(feature_dict, json_entry, state_json):
+        feature_dict[AllFeature.parent].append({AllFeature.name: json_entry})
+    
+
+FEATURE_LIST = [AllFeature]
+
+
+def get_feature_dictionary(log_file, feature_list, state_json):
+    '''
+    Extracts features from the journal entries and places them in a dictionary.
+
+    :param log_file: iterator of journal entries
+    :param feature_list: comma-separated list of feature names to extract
+    :return: dictionary of features
+    :raises: ValueError if an invalid feature name is provided
+    '''
+    feature_list = feature_list.split(",")
+
+    feature_dict = defaultdict(list)
+    feature_classes = [cls for cls in FEATURE_LIST
+                       if cls.name in feature_list]
+    if len(feature_classes) != len(feature_list):
+        raise ValueError(
+            "Error: Invalid feature name in feature list {}".format(feature_list))
+
+    for line in log_file.readlines():
+        try:
+            line_json = json.loads(line)
+            for feature_class in feature_classes:
+                feature_class.maybe_add_feature(feature_dict, line_json, state_json)
+        except json.JSONDecodeError:
+            pass
```

Comment:
maybe we should raise an error here? we can later make this a warning or a log message, but early on I'd rather fail to root out all the issues